### PR TITLE
feat: use default global client when no use option

### PR DESCRIPTION
**Pull request by li-jin-gou**
#### What type of PR is this?

feat

#### What this PR does / why we need it (English/Chinese):

zh: 当没有使用 option 时使用默认client
en: use default global client when no use option

### CODE REVIEW

File: reverse_proxy.go [hunk 1]
```diff
@@ -97,6 +97,10 @@ var hopHeaders = []string{
 // NewSingleHostReverseProxy does not rewrite the Host header.
 // To rewrite Host headers, use ReverseProxy directly with a custom
 // director policy.
+//
+// Note: it uses the default global client.Client when you do not pass config.ClientOption.
+// When using pass config.ClientOption it initializes a local client.Client.
```
Comment (welkeyever) [inline reverse_proxy.go]: using pass -> passing

Comment (welkeyever) [reply reverse_proxy.go]: it initializes a -> it will initialize a

Comment (welkeyever) [reply reverse_proxy.go]: client.Client -> client.Client instance

Comment (li-jin-gou) [reply reverse_proxy.go]: done

File: reverse_proxy.go [hunk 2]
```diff
@@ -97,6 +97,10 @@ var hopHeaders = []string{
 // NewSingleHostReverseProxy does not rewrite the Host header.
 // To rewrite Host headers, use ReverseProxy directly with a custom
 // director policy.
+//
+// Note: it uses the default global client.Client instance when you do not pass config.ClientOption.
```
Comment (welkeyever) [inline reverse_proxy.go]: it uses the -> it will use the

Comment (welkeyever) [inline reverse_proxy.go]: pass config -> pass any config

Comment (li-jin-gou) [reply reverse_proxy.go]: done

Comment (li-jin-gou) [reply reverse_proxy.go]: done

File: reverse_proxy.go [hunk 3]
```diff
@@ -97,6 +97,10 @@ var hopHeaders = []string{
 // NewSingleHostReverseProxy does not rewrite the Host header.
 // To rewrite Host headers, use ReverseProxy directly with a custom
 // director policy.
+//
+// Note: it will use the default global client.Client instance when you do not pass any config.ClientOption.
```
Comment (welkeyever) [inline reverse_proxy.go]: when you do not pass any config.ClientOption. -> if no config.ClientOption is passed.

File: reverse_proxy.go [hunk 4]
```diff
@@ -97,6 +97,10 @@ var hopHeaders = []string{
 // NewSingleHostReverseProxy does not rewrite the Host header.
 // To rewrite Host headers, use ReverseProxy directly with a custom
 // director policy.
+//
+// Note: if no config.ClientOption is passed it will use the default global client.Client instance.
+// When passing config.ClientOption it will initialize a local client.Client instance.
+// If need to customize client.Client instance then use ReverseProxy.SetClient.
```
Comment (welkeyever) [inline reverse_proxy.go]: Using ReverseProxy.SetClient if there is need for customized client.Client instance.

Comment (li-jin-gou) [reply reverse_proxy.go]: done

File: reverse_proxy.go (hunk 3 above)
Comment (li-jin-gou) [reply reverse_proxy.go]: done

File: reverse_proxy.go [hunk 5]
```diff
@@ -97,6 +97,10 @@ var hopHeaders = []string{
 // NewSingleHostReverseProxy does not rewrite the Host header.
 // To rewrite Host headers, use ReverseProxy directly with a custom
 // director policy.
+//
+// Note: if no config.ClientOption is passed it will use the default global client.Client instance.
+// When passing config.ClientOption it will initialize a local client.Client instance.
+// Using ReverseProxy.SetClient if there is need for customized client.Client instance.
```
Comment (welkeyever) [inline reverse_proxy.go]: need for customized client.Client -> need for shared customized client.Client

Comment (li-jin-gou) [reply reverse_proxy.go]: done

**Review by welkeyever — APPROVED** [review]

**Pull request merged**